File: frontend/src/types/notification.ts
import { User } from './user'

export enum NotificationType {
  SYSTEM = 'SYSTEM',
  COURSE = 'COURSE',
  PAYMENT = 'PAYMENT',
  GENERAL = 'GENERAL'
}

export enum TargetRole {
  ALL = 'ALL',
  ADMIN = 'ADMIN',
  TEACHER = 'TEACHER',
  STUDENT = 'STUDENT'
}

export interface Notification {
  id: number
  title: string
  content?: string
  type: NotificationType
  targetRole: TargetRole
  isPublished: boolean
  publishedAt?: string
  createdBy?: User
  createdAt: string
  updatedAt: string
}

export interface NotificationRequest {
  title: string
  content?: string
  type?: NotificationType
  targetRole?: TargetRole
  isPublished?: boolean
}

File: backend/src/main/java/com/example/backend/entity/Notification.java
package com.example.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "notifications")
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "通知标题不能为空")
    @Column(nullable = false, length = 100)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Type type = Type.GENERAL;

    @Enumerated(EnumType.STRING)
    @Column(name = "target_role", length = 20)
    private TargetRole targetRole = TargetRole.ALL;

    @Column(name = "is_published")
    private Boolean isPublished = false;

    @Column(name = "published_at")
    private LocalDateTime publishedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum Type {
        SYSTEM, COURSE, PAYMENT, GENERAL
    }

    public enum TargetRole {
        ALL, ADMIN, TEACHER, STUDENT
    }
}
